src/client/Health/NodeView.jsx

```diff
@@ -25,37 +37,232 @@ const height = 500
+    history.push({ pathname })
+  }
+
+  // End Tools *******************************
 
   const northNodes = applications?.map((application, index) => (
+
     {
       id: application.applicationId, // unique id of node
       type: 'output', // output node
       targetPosition: 'bottom', // handle is at the bottom
       style: {
-        background: colors.background[application.enabled ? 'enabled' : 'disabled'],
-        border: colors.border[application.enabled ? 'enabled' : 'disabled'],
-        width: '140px',
+        background: colors.background[application.enabled ? 'disabled' : 'disabled'],
+        border: colors.border[application.enabled ? 'disabled' : 'disabled'],
+        width: '170px',
+        height: '25%',
       },
       data: {
         label: (
-          <div id={`north-${index}`} className={`oi-box tight text-${application.enabled ? 'success' : 'muted'}`}>
-            <Link to={`/north/${application.applicationId}`}>
+          <div className="box-container">
+            <div className="icon-container">
+              <div className="icon-left">
+                <img src={`${ApiSchemas[application.api].image}`} alt="logo" height="20px" />
+              </div>
+              <div className="icon-center flex-grow">
+                {`${application.api}`}
+              </div>
+              <div className={`icon-right ${application.enabled ? 'icon-status-enabled' : 'icon-status-disabled'}`} />
+            </div>
+
+            <div id={`north-${index}`} className={` oi-box tight text-${application.enabled ? 'muted' : 'muted'}`}>
               <div style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
-                {application.applicationId}
+                <EditableIdField
+                  id={application.applicationId}
+                  fromList={sortableApplications}
+                  index={index}
+                  name="applicationId"
+                  valid={validation.application.isValidName}
+                  idChanged={handleApplicationIdChanged}
+                />
               </div>
               <UncontrolledTooltip placement="top" target={`north-${index}`} innerClassName="oi-pop">
                 {application.applicationId}
               </UncontrolledTooltip>
-              <div>{`(${application.api})`}</div>
-            </Link>
+            </div>
+
+            <div className="icon-settings">
+              <FontAwesomeIcon icon={faEllipsisV} id="NorthOption" />
+              <UncontrolledPopover trigger="legacy" placement="bottom" target="NorthOption">
+                <PopoverHeader> North Options</PopoverHeader>
+                <PopoverBody className="icon-pop">
+                  <Button
+                    className="icon-pop-items"
+                    type="button"
+                    onClick={(e) => {
+                      e.preventDefault()
+                      handleEditNorth(index)
+                    }}
+                  >
```

Comment:
The problem here is that the onClick method is called when the user click on the button. index is interpolated at this moment, when the forEach has been done at page load. So the index value kept in memory is the last one